datafusion/execution/src/cache/cache_manager.rs

```diff
@@ -32,8 +33,27 @@ use std::sync::Arc;
 /// session lifetime.
 ///
 /// See [`crate::runtime_env::RuntimeEnv`] for more details
-pub type FileStatisticsCache =
-    Arc<dyn CacheAccessor<Path, Arc<Statistics>, Extra = ObjectMeta>>;
+pub trait FileStatisticsCache:
+    CacheAccessor<Path, Arc<Statistics>, Extra = ObjectMeta>
+{
+    /// Retrieves the information about the entries currently cached.
+    fn list_entries(&self) -> HashMap<Path, FileStatisticsCacheEntry>;
```

Comment:
Do we want to make this thread safe? When we bring in size limit, we might want to evict the cache object as well

Reply:
Not sure I understand, this should be thread safe as is right?